application/src/main/java/org/togetherjava/tjbot/commands/moderation/report/ReportCommand.java

```diff
@@ -0,0 +1,175 @@
+        reportedMessage = event.getTarget().getContentRaw();
+        reportedMessageUrl = event.getTarget().getJumpUrl();
+
+        TextInput body = TextInput.create(MESSAGE_INPUT, "Message", TextInputStyle.PARAGRAPH)
+            .setPlaceholder("Reason for reporting")
+            .setRequiredRange(10, 200)
+            .build();
+
+        Modal modal = Modal.create(generateComponentId(), "Report").addActionRow(body).build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long userID = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), guildID);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+            return;
+        }
+        String modalMessage = event.getValue(MESSAGE_INPUT).getAsString();
+        MessageCreateAction message =
+                createModMessage(modalMessage, userID, modMailAuditLog.orElseThrow());
+
+        sendMessage(event, message);
+
+        event.getHook().editOriginal("Thank you for your feedback!").queue();
+
+    }
+
+    private boolean handleIsOnCooldown(long userID, ModalInteractionEvent event) {
+        if (!isChannelOnCooldown(userID)) {
+            return false;
+        }
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isChannelOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private Optional<TextChannel> getModMailChannel(JDA jda, long guildID) {
+        return jda.getGuildById(guildID)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private MessageCreateAction createModMessage(String modalMessage, long userID,
+            TextChannel modMailAuditLog) {
+        return modMailAuditLog.sendMessageEmbeds(createReportMessage(modalMessage, userID));
+    }
+
+    private void sendMessage(ModalInteractionEvent event, MessageCreateAction message) {
```

Comment:
the responsibility on this feels weird. i would expect that it creates the message itself, rather than getting it as argument. i would change the flow slightly.

Reply:
How would you change the flow?

Reply:
as i said. i would let this method create the message and send it. instead of just sending it, having the creation happen elsewhere.

Reply:
Done